### PR TITLE
Add and use `fetch.max.bytes` to limit total Fetch response size (KIP-74, #1616)

### PR DESCRIPTION
**Pull request by edenhill**
`receive.message.max.bytes` was used in FetchRequest.max_bytes to tell
the broker to limit the response size, but since the broker may overshoot
this value up to one message.max.bytes it resulted in an endless config
chase where receive.message.max.bytes was increased and the response
size would follow + a couple of extra bytes.
Using the new `fetch.max.bytes` property and making sure it is lower than
`receive.message.max.bytes` fixes the issue.

This also adds verification for these configuration values to
the rd_kafka_new() call.

### CODE REVIEW

**Comment by ijuma** [comment]
To be clear, the broker will only exceed max bytes if a single batch is larger than the limit (i.e. it would need return two batches where the second would cause it to go over). Given that, it seems like a config bug is needed to trigger this issue. I don't quite understand how this PR fixes that. It would be good to talk about a specific example of how it would break previously and now would work.

**Comment by edenhill** [comment]
This is a typical error message from this case:
`Receive failed: Invalid response size 1000000040 (0..1000000000): increase receive.message.max.bytes`

receive.message.max.bytes is set to 1000000000, and reused as FetchRequest.max_bytes, the response has the overhead of FetchResponse framing and thus exceeds receive.message.max.bytes.
The problem typically surfaces on high-volume topics when fetching multiple partitions and there is no easy workaround.

File: src/rdkafka_broker.c [hunk 1]
```diff
@@ -3038,7 +3038,7 @@ static int rd_kafka_broker_fetch_toppars (rd_kafka_broker_t *rkb, rd_ts_t now) {
         if (rd_kafka_buf_ApiVersion(rkbuf) == 4) {
                 /* MaxBytes */
                 rd_kafka_buf_write_i32(rkbuf,
-                                       rkb->rkb_rk->rk_conf.recv_max_msg_size);
+                                       rkb->rkb_rk->rk_conf.fetch_max_bytes);
```
Comment (edenhill) [inline src/rdkafka_broker.c]: This is the essence of the fix, all the rest is config property handling.

**Comment by mhowlett** [comment]
Wondering if there enough value in the extra configuration property to justify adding it? (making the library overly tunable acts to make it harder to use). Can you just use a const internal to librdkafka corresponding to the protocol overhead instead? 

If you definitely believe the extra config property is justified, I think the doc string could do with a bit of re-wording to really highlight the difference with `receive.message.max.bytes` (and implicitly justify it's existence).

**Comment by ijuma** [comment]
Yeah, if it's just the framing issue, we should consider whether we can make it work without an additional config.

**Comment by edenhill** [comment]
Note that fetch.max.bytes is the new config property, receive.message.max.bytes on the other is an existing property that we can't remove - in hindsight there might be little need for it.

fetch.max.bytes does have some use for people with certain performance requirements or unusually large messages.

**Comment by edenhill** [comment]
Will reconstruct PR as to other external discussion:
 * automatically adjust receive.message.max.bytes to be at least fetch.max.bytes + framing
 * remove reference to receive.message.max.bytes in fetch.max.bytes config doc to avoid confusion.

**Comment by mhowlett** [comment]
LGTM

**Pull request merged**